### PR TITLE
(test) O3-4165: Add tests for lab test panel view

### PR DESCRIPTION
**Pull request by ODORA0**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary

Adds test coverage for the Test Panel view in the Results Viewer.

## Related Issue
https://openmrs.atlassian.net/browse/O3-4165

### CODE REVIEW

File: packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  }),
+}));
+
+jest.mock('@openmrs/esm-framework', () => ({
+  formatDate: jest.fn().mockReturnValue('January 1, 2024'),
+  isDesktop: jest.fn().mockReturnValue(true),
+  useLayoutType: jest.fn().mockReturnValue('desktop'),
+}));
+
+jest.mock('./helper', () => ({
```
Comment (CynthiaKamau) [inline packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx:21]: Thanks @ODORA0 , whats the use case for this ?

File: packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        observations={mockObservations}
+        activePanel={null}
+        setActivePanel={mockSetActivePanel}
+      />,
+    );
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Complete Blood Count/i,
+    });
+    fireEvent.click(buttonElement);
```
Comment (CynthiaKamau) [inline packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx:159]: Could we use userEvent instead ?

File: packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        panel={mockBasePanel}
+        observations={mockObservations}
+        activePanel={null}
+        setActivePanel={mockSetActivePanel}
+      />,
+    );
+
+    expect(screen.getByRole('table')).toHaveClass('cds--data-table--sm');
+
+    (useLayoutType as unknown as jest.Mock).mockReturnValue('tablet');
```
Comment (CynthiaKamau) [inline packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx:275]: const mockUseLayoutType = jest.mocked(useLayoutType);
mockUseLayoutType.mockReturnValue('tablet');

File: packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    (isDesktop as unknown as jest.Mock).mockReturnValue(false);
```
Comment (CynthiaKamau) [inline packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx:276]: same here

File: packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx (hunk 2 above)
Comment (ODORA0) [reply packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx]: Done

File: packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx [hunk 5]
```diff
@@ -0,0 +1,280 @@
+import React from 'react';
+import { render, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import LabSetPanel from './panel.component';
+import { isDesktop, useLayoutType, type LayoutType } from '@openmrs/esm-framework';
+import { type ObsRecord } from '../../types';
+import { type OBSERVATION_INTERPRETATION } from '@openmrs/esm-patient-common-lib';
+
+jest.mock('react-i18next', () => ({
```
Comment (CynthiaKamau) [inline packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx:9]: remove this

**Review by CynthiaKamau — COMMENTED** [review]
Thanks @ODORA0 , left a comment

File: packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+import LabSetPanel from './panel.component';
+import { isDesktop, useLayoutType, type LayoutType } from '@openmrs/esm-framework';
+import { type ObsRecord } from '../../types';
+import { type OBSERVATION_INTERPRETATION } from '@openmrs/esm-patient-common-lib';
+
+jest.mock('@openmrs/esm-framework', () => ({
+  formatDate: jest.fn().mockReturnValue('January 1, 2024'),
+  isDesktop: jest.fn().mockReturnValue(true),
+  useLayoutType: jest.fn().mockReturnValue('desktop'),
+}));
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx:13]:
```suggestion
```

We don't need to override the partial framework mock here https://o3-docs.openmrs.org/docs/frontend-modules/unit-and-integration-testing#you-should-not-need-to-use-partial-mocks.

**Review by CynthiaKamau — APPROVED** [review]

File: packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+  },
+  effectiveDateTime: '2024-01-01T10:00:00Z',
+  issued: '2024-01-01T10:00:00Z',
+  name: 'Complete Blood Count',
+  value: '120',
+  interpretation: 'NORMAL',
+  relatedObs: [],
+  meta: mockConceptMeta,
+  referenceRange: [],
+};
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx:65]: If these mocks are meant to be used in more than one place, it'd be better to move them to the top-level `__mocks__` directory.

File: packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+  },
+];
+
+describe('LabSetPanel', () => {
+  const mockSetActivePanel = jest.fn();
+  const user = userEvent.setup();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/test-results/panel-view/panel.component.test.tsx:100]: Jest does this automatically, so you don't need to clear mocks by hand.

**Review by denniskigen — COMMENTED** [review]
Thanks for working on this, @ODORA0. It's close, just a few tweaks needed to get it over the line.

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks, @ODORA0.

**Pull request merged**